src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBinderSourceGenerator.Emitter.cs

```diff
@@ -0,0 +1,469 @@
+            private const string NotSupportedReadon_OnlyComplexObjectsSupported = "Only object or collection types are supported";
+            private const string NotSupportedReason_TypeNotDetectedAsInput = "Generator parser did not detect the type as input";
+
+            private readonly ConfigurationBinderSourceGenerationContext _sourceGenerationContext;
+            private readonly Parser.InputTypes _targetTypes;
+
+            private readonly Queue<ITypeSymbol> _privateBindMethodGen_QueuedTypes = new();
+#pragma warning disable RS1024
+            private readonly HashSet<ITypeSymbol> _bindMethodGen_ProcessedTypes = new(SymbolEqualityComparer.Default);
+            private readonly HashSet<ITypeSymbol> _unsupportedTypes = new(SymbolEqualityComparer.Default);
+#pragma warning restore RS1024
+
+            private readonly INamedTypeSymbol? _typeSymbolForIConfigurationSection;
+
+            // Collections
+            private readonly INamedTypeSymbol _typeSymbolForIEnumerable;
+            private readonly INamedTypeSymbol? _typeSymbolForGenericIEnumerable;
+
+            private readonly INamedTypeSymbol? _typeSymbolForGenericIDictionary;
+            private readonly INamedTypeSymbol? _typeSymbolForGenericDictionary;
+
+            // Postfix for stringValueX variables used to save config value indexer
+            // results e.g. if (configuration["Key"] is string stringValue0) { ... }
+            private int _parseValueCount;
+
+            private readonly SourceWriter _writer = new();
+
+            public Emitter(ConfigurationBinderSourceGenerationContext sourceGenerationContext, Parser.InputTypes targetTypes)
+            {
+                _sourceGenerationContext = sourceGenerationContext;
+                _targetTypes = targetTypes;
+
+                Compilation compilation = _sourceGenerationContext.Compilation;
+                _typeSymbolForIConfigurationSection = compilation.GetBestTypeByMetadataName(FullNameForIConfigurationSectionType);
+                _typeSymbolForIEnumerable = compilation.GetSpecialType(SpecialType.System_Collections_IEnumerable);
+                _typeSymbolForGenericIEnumerable = compilation.GetSpecialType(SpecialType.System_Collections_Generic_IEnumerable_T)?.ConstructUnboundGenericType();
+                _typeSymbolForGenericIDictionary = compilation.GetBestTypeByMetadataName(FullNameForGenericIDictionary)?.ConstructUnboundGenericType();
+                _typeSymbolForGenericDictionary = compilation.GetBestTypeByMetadataName(FullNameForGenericDictionary)?.ConstructUnboundGenericType();
+            }
+
+            public void Emit()
+            {
+                const string LiteralFor_GeneratedConfigurationBinder = "GeneratedConfigurationBinder";
+
+                _writer.WriteLine(@"// <auto-generated/>
+#nullable enable annotations
+#nullable disable warnings
```

Comment:
>nullable [](http://example.com/codeflow?start=1&length=8)

do we usually disable the nullable warning?

Reply:
We're currently doing the same for the JSON generator - https://github.com/dotnet/runtime/pull/74801. Since this is a fresh generator, we can try to do the right thing & enable it with `#nullable enable`